content/working_with/official_plugins/Orchestration/kubernetes.md

```diff
@@ -397,3 +485,175 @@ cfy executions start update_resource_definition -d pod -vv -p resource_definitio
   }
 }" -p node_instance_id=nginx_9pqgdu
 {{< /highlight >}}
+
+# Using the kubernetes plugin with EKS, AKS, GKE
+The kubernetes plugin works with EKS, AKS, GKE.
+On this section we will demonstrate how to create a kubernetes cluster on each of them using their respective plugin(AWS/GCP/Azure) alongside kubernetes plugin. 
+
+## EKS cluster
+
+On our examples repository, there is an example of [deploying an EKS cluster.] (https://github.com/cloudify-community/blueprint-examples/blob/master/kubernetes/aws-eks/blueprint.yaml)
+
+this example demonstrates a deployment of eks cluster with one node group.
+We will explain how we used the AWS plugin alongside kubernetes plugin on this example in order to deploy the cluster.
+
+Firstly, in order to create an eks cluster we need to create all the aws node_instances that needed like:
+vpc, subnets , security group, etc. 
+Then, we create the eks cluster and save the kubeconfig content in its runtime properties.
+you can read more about those aws instances in our [AWS plugin documentation]({{< relref "working_with/official_plugins/Infrastructure/aws.md" >}}).
+
+After we have the eks cluster and the node group deployed, we will use the kubernetes plugin to generate the service account authentication token
+(creating service account and rule binding) and create a sanity_master node instance:
```

Comment:
`The blueprint defines service account and token node types whose purpose is to generate a token that can be used for simpler plugin authentication.`